iothub_client/samples/pnp/pnp_simple_thermostat/pnp_simple_thermostat.c

```diff
@@ -365,270 +327,245 @@ static void UpdateTemperatureAndStatistics(double desiredTemp, bool* maxTempUpda
     {
-        LogError("Cannot read environment variable=%s", g_dpsDeviceKeyEnvironmentVariable);
-        result = false;
+        LogError("IoTHubMessage_SetContentEncodingSystemProperty failed, error=%d", messageResult);
     }
-    else
+    // Send the telemetry message.
+    else if ((iothubClientResult = IoTHubDeviceClient_LL_SendTelemetryAsync(deviceClient, messageHandle, NULL, NULL)) != IOTHUB_CLIENT_OK)
     {
-        g_pnpDeviceConfiguration.securityType = PNP_CONNECTION_SECURITY_TYPE_DPS;
-        result = true;    
+        LogError("Unable to send telemetry message, error=%d", iothubClientResult);
     }
 
-    return result;
-#endif // USE_PROV_MODULE_FULL
+    IoTHubMessage_Destroy(messageHandle);
 }
 
-
 //
-// GetConfigurationFromEnvironment reads how to connect to the IoT Hub (using 
-// either a connection string or a DPS symmetric key) from the environment.
+// CreateDeviceClientLLHandle does the creates the IOTHUB_DEVICE_CLIENT_LL_HANDLE based on environment configuration.
+// If PNP_CONNECTION_SECURITY_TYPE_DPS is used, the call will block until DPS provisions the device.
 //
-static bool GetConnectionSettingsFromEnvironment()
+static IOTHUB_DEVICE_CLIENT_LL_HANDLE CreateDeviceClientLLHandle(void)
 {
-    const char* securityTypeString;
-    bool result;
+    IOTHUB_DEVICE_CLIENT_LL_HANDLE deviceClient = NULL;
 
-    if ((securityTypeString = getenv(g_securityTypeEnvironmentVariable)) == NULL)
-    {
-        LogError("Cannot read environment variable=%s", g_securityTypeEnvironmentVariable);
-        result = false;
-    }
-    else
+    if (g_pnpDeviceConfiguration.securityType == PNP_CONNECTION_SECURITY_TYPE_CONNECTION_STRING)
     {
-        if (strcmp(securityTypeString, g_securityTypeConnectionStringValue) == 0)
-        {
-            result = GetConnectionStringFromEnvironment();
-        }
-        else if (strcmp(securityTypeString, g_securityTypeDpsValue) == 0)
-        {
-            result = GetDpsFromEnvironment();
-        }
-        else
+        if ((deviceClient = IoTHubDeviceClient_LL_CreateFromConnectionString(g_pnpDeviceConfiguration.u.connectionString, MQTT_Protocol)) == NULL)
         {
-            LogError("Environment variable %s must be either %s or %s", g_securityTypeEnvironmentVariable, g_securityTypeConnectionStringValue, g_securityTypeDpsValue);
-            result = false;
+            LogError("Failure creating IotHub client.  Hint: Check your connection string");
         }
     }
-
-    return result;    
-}
-
-//
-// CreateDeviceClientLLHandle performs actual handle creation (but nothing more), depending
-// on whether connection strings or DPS is used.
-//
-static IOTHUB_DEVICE_CLIENT_LL_HANDLE CreateDeviceClientLLHandle(void)
-{
 #ifdef USE_PROV_MODULE_FULL
-    if (g_pnpDeviceConfiguration.securityType == PNP_CONNECTION_SECURITY_TYPE_DPS)
+    else if ((deviceClient = PnP_CreateDeviceClientLLHandle_ViaDps(&g_pnpDeviceConfiguration)) == NULL)
```

Comment:
doesn't DPS have a connection string scenario? Would the line on 556 be invoked if someone sets a DPS connection string?

Reply:
By convention across samples, if $ENV:IOTHUB_DEVICE_SECURITY_TYPE == "connectionString" this implies IoT Hub connection string.  DPS is if the env == "DPS".  

I suppose we could've been more explicit with "hubConnectionString" but as all SDKs (except embedded which has its own nomenclature around AZ_* stuff) and existing customers already are on connectionString we're on that.